checklist.md

```diff
@@ -0,0 +1,53 @@
+- [ ] **Hook Scripts** - all hook scripts (pre-/post- provisioning and deployment scripts) shall include both sh and pwsh versions.
+ 
+## Functional requirements
+ 
+- [ ] **azd up** - successfully provisions and deploys a functional app with `azd up`
+- [ ] **azd Pipeline Config** - `.github/workflows` should include [`azure-dev.yaml`](https://github.com/Azure-Samples/todo-python-mongo-aca/blob/main/.github/workflows/azure-dev.yml) to support `azd pipeline config`
+- [ ] **GitHub Actions** run tasks without errors
+- [ ] **DevContainer** has been tested locally and runs
+- [ ] **Codespaces** run [locally and in browser]
+- [ ] **Dashboard** - Resources should include a dashboard so that `azd monitor` works, either by referencing the `monitoring.bicep` module or creating a dashboard separately. See [main.bicep](https://github.com/Azure-Samples/todo-python-mongo-aca/blob/05a7aa59c0253628e293ca0fcc98f35a942df1cc/infra/main.bicep#L129)
+    - [ ] **Monitoring** - Application code should include either OpenCensus or OpenTelemetry so that the monitor is populated. See [todo/app.py](https://github.com/Azure-Samples/todo-python-mongo-aca/blob/05a7aa59c0253628e293ca0fcc98f35a942df1cc/src/api/todo/app.py#L58)
+- [ ] All tests pass
+    
+    In the absense of e2e tests,
+    - [ ] The application has been manually tested to work as per the requirement
+ 
+## Security requirements
+ 
+- [ ] **Managed Identity** - Microsoft Managed Service Identity is implemented
+- [ ] **security-devops-action** - The application must run microsoft/security-devops-action. [Example](https://github.com/Azure-Samples/azure-search-openai-demo/blob/ab1e4806a176b084b1980a9ee7e1c55bdac1a6d5/.github/workflows/azure-dev-validation.yaml#L25)
+    - If unfamiliar with setting up GH Actions, then you can follow the instructions here: [defender-for-cloud/quickstart-onboard-github](https://learn.microsoft.com/en-us/azure/defender-for-cloud/quickstart-onboard-github)
+ 
+When a service selected doesn't support Managed Identity, the corresponding issue must have been reported and the security considerations section in the readme, should clearly explain the alternatives.
+ 
+- Azure Key Vault is a preferred alternative
+ 
+### The following items are not strictly enforced but may prevent the template from being added to the gallery
+ 
+- [ ] Project code follows standard structure, [per language](../structure-samples/structure-samples.md)
+- [ ] Code follows recommended styleguide
```

Comment:
Do we have a link/links we can point them to?

Reply:
we can point to the styleguides from Natalia https://github.com/Azure-Samples/azd-template-artifacts/pull/3/files once her pr is merged!